src/codeflare_sdk/cluster/config.py

```diff
@@ -206,4 +203,24 @@ def _memory_to_resource(self):
             warnings.warn("max_memory is being deprecated, use worker_memory_limits")
             self.worker_memory_limits = f"{self.max_memory}G"
 
-    local_queue: str = None
+    def validate_types(self):
+        """Validate the types of all fields in the ClusterConfiguration dataclass."""
+        for field_info in fields(self):
+            value = getattr(self, field_info.name)
+            expected_type = field_info.type
+            if not self.is_valid_type(value, expected_type):
+                raise TypeError(
+                    f"'{field_info.name}' should be of type {expected_type}, got {type(value).__name__}"
+                )
+
+    @staticmethod
+    def is_valid_type(value, expected_type):
```

Comment:
Small nit for function name
```suggestion
    def _is_type(value, expected_type):
```